src/main/java/org/xbill/DNS/Zone.java

```diff
@@ -65,7 +69,7 @@ public boolean hasNext() {
     }
 
     @Override
-    public RRset next() {
+    public RRset next() throws IllegalMonitorStateException {
```

Comment:
Is the exception declaration really necessary? I'm hesitant to add this to the public API.

Reply:
If we move the locking and unlocking out of the try/finally block, then yes, this exception will be necessary.